lib/components/SegmentedControl/segmented-control.scss

```diff
@@ -0,0 +1,167 @@
+            }
+            &:disabled {
+                opacity: var(--core-opacity-300);
+                cursor: not-allowed;
+                pointer-events: none;
+            }
+            &.focused,
+            &.focused::before {
+                // not using default :focus since we are customizing focus state
+                box-shadow: 0 0 0 var(--borderWidth-xs)
+                    var(--core-color-solid-blue-700) inset;
+            }
+            &:hover:not(:disabled) {
+                background: color-mix(
+                    in srgb,
+                    var(--semantic-color-typography-prominent) 8%,
+                    transparent
+                );
+            }
+            &:active:not(:disabled, .selected),
+            &:active:not(:disabled, .animated)::before {
+                box-shadow: unset;
+                background: color-mix(
+                    in srgb,
+                    var(--semantic-color-typography-prominent) 16%,
+                    transparent
+                );
+            }
+            &.selected:not(:disabled) {
+                // colors should stay the same for both themes:
+                background: var(--core-color-solid-slate-50);
+            }
+            &.animated::before {
+                content: "";
+                position: absolute;
+                background-color: var(--core-color-solid-slate-50);
+                width: 100%;
+                height: 100%;
+                animation: var(--core-motion-duration-200)
+                    var(--motion-easing-inandout) slide-in;
+                top: 0;
+                left: 0;
+                border-radius: inherit;
+                z-index: 1;
+            }
+            .icon {
+                display: flex;
+                align-items: center;
+                justify-content: center;
+
+                svg {
+                    fill: var(--semantic-color-typography-default);
+                }
+            }
+            &.selected:not(:disabled),
+            &.animated {
+                // colors should stay the same for both themes:
+                * {
+                    color: var(--core-color-solid-slate-1400);
+                }
+                svg {
+                    fill: var(--core-color-solid-slate-1400);
+                }
+            }
+        }
+        &--has-container-width {
+            width: inherit;
+            min-width: fit-content;
+
+            .item {
+                flex: 1;
+            }
+        }
+        &--sm {
+            height: var(--size-generic-lg);
+            padding: var(--semantic-spacing-general-xs);
+            border-radius: var(--semantic-borderRadius-md);
+
+            .item {
+                height: var(--size-generic-md);
+                padding: var(--temp-static-spacing-5)
+                    var(--semantic-spacing-general-sm)
+                    var(--temp-static-spacing-5)
+                    var(--semantic-spacing-general-sm);
```

Comment:
I will merge this but let's minimize our css as much as possible. Please apply the fix using a new PR. 
```suggestion
```